sdk/eventhub/event-hubs/src/eventHubConsumerClient.ts

```diff
@@ -309,7 +316,16 @@ export class EventHubConsumerClient {
    * @returns Promise<void>
    * @throws Error if the underlying connection encounters an error while closing.
    */
-  close(): Promise<void> {
+  async close(): Promise<void> {
+    // Stop all the actively running subscriptions.
+    const activeEventProcessors = Array.from(this._eventProcessors);
+    await Promise.all(
```

Comment:
Should we try/catch around this? A failure in one processor to close shouldn't prevent us from closing the connection.

Reply:
I went back and forth on this, and the reason I decided to let this throw was because then the user could see that a subscription failed to be closed and that's why the client didn't close. Otherwise I'm worried we get stuck in the same situation we're trying to fix...where an open subscription keeps the process open even though close was called on the client.

With that said, do you still think we should just catch and log the error instead?

Reply:
That's very reasonable.

Perhaps we could catch, close the connection and then throw the error? So at least then they see the error but the greater operation (of making sure the connection gets closed) still happens?

Reply:
What if subscriptions fail to close, and we fail to close the connection, which error do we show then?

In either scenario, there may be something blocking the process from exiting, so I'd recommend that the user retry calling `close()`. With that in mind, I'm not sure there's value in continuing the operation if we encounter an error closing a subscription.

Reply:
It seems like the connection one should take precedence there  but obviously that's just arbitrary.

My main concern is the user will not be able to close _ever_ because one of the subscriptions is failing to close. I'm assumign this "close all the other things for you" is motivated by people that aren't tracking individual subscriptions or doing the hygenic "close the subscriptions yourself" behavior.

In which case I'd expect best-effort on those but the primary option if shutting down the connection should always work.

Reply:
Just adding - I don't think this is a "don't merge" concern and there is certainly the greater concern of "user can't even exit the app because subscriptions aren't cleaning up". So please feel free to merge if this is all that's left.